src/Bicep.Local.Deploy/LocalDeploymentEngine.cs

```diff
@@ -0,0 +1,189 @@
+
+            var deploymentPlan = await this.deploymentEngine.ProcessDeployment(
+                preflightSettings: new(settings, syncMode: true),
+                deploymentContext: context,
+                definition: definition,
+                originalDeploymentName: context.DeploymentName,
+                isNestedDeployment: false,
+                validateOnly: false,
+                request: null,
+                frontdoorEndpointUri: new Uri(requestContext.FrontdoorEndpoint),
+                cancellationToken: cancellationToken,
+                oboToken: oboToken,
+                oboCorrelationId: oboCorrelationId);
+
+            await StartDeployment(deploymentPlan);
+
+            var dataProvider = dataProviderHolder.GetDeploymentDataProvider(deploymentPlan.DeploymentLocation);
+            return await WaitForDeploymentToComplete(dataProvider, context, cancellationToken);
+        }
+    }
+
+    private async Task<LocalDeployment.Result> WaitForDeploymentToComplete(IDeploymentDataProvider dataProvider, DeploymentContext context, CancellationToken cancellationToken)
+    {
+        var entity = await dataProvider.FindDeployment(context.SubscriptionId, context.ResourceGroupName, context.DeploymentName);
+        while (!entity.ProvisioningState.IsTerminal())
+        {
+            await Task.Delay(500, cancellationToken);
+
+            entity = await dataProvider.FindDeployment(context.SubscriptionId, context.ResourceGroupName, context.DeploymentName);
+        }
+
+        var operationEntities = await dataProvider.FindDeploymentOperations(context.SubscriptionId, context.ResourceGroupName, context.DeploymentName, entity.SequenceId, -1);
+
+        return new(
+            deploymentEngine.CreateDeploymentDefinition(entity, requestContext.ApiVersion),
+            operationEntities
+                .Where(operation => operation.TargetResource is not null)
+                .Select(operation => deploymentEngine.CreateDeploymentOperationDefinition(entity, operation)).ToImmutableArray());
+    }
+
+    private async Task StartDeployment(DeploymentPlan deploymentPlan)
+    {
+        try
+        {
+            await jobProvider.CreateDeploymentJob(
+                deploymentJob: deploymentPlan.DeploymentJob,
+                onCommitJobDefinitionDelegate: () => SaveDeployment(deploymentPlan)).ConfigureAwait(false);
+        }
+        catch (JobSizeExceededException exception)
+        {
+            var message = ErrorResponseMessages.DeploymentJobSizeExceeded.ToLocalizedMessage();
+
+            throw new ErrorResponseMessageException(
+                httpStatus: HttpStatusCode.BadRequest,
+                errorCode: DeploymentsErrorResponseCode.DeploymentSizeExceeded,
+                errorMessage: message,
+                innerException: exception);
+        }
+    }
+
+    private async Task SaveDeployment(DeploymentPlan deploymentPlan)
```

Comment:
Just to clarify, SaveDeployment means that the deployment is kept track of locally, correct?

Reply:
I'm using in-memory data providers, so when we "save" an entity, we are replacing it in memory. This is only available while the process is running - it is not persisted afterwards.